### PR TITLE
feat: copy changes before payments is released to prod

### PR DESCRIPTION
**Pull request by LinHuiqing**
## Problem

Agencies onboarding to payments require our hand-holding, but this is not sustainable in the long run as we want to scale.

Closes FRM-893, closes FRM-439 (opened new issues for remaining point and discrepancies)

## Solution

**Breaking Changes** 

- [x] No - this PR is backwards compatible  

**Features**:

- Announcement rollout for payments (FRM-439)
- What's new section for payments (FRM-439)
- Add Supports payments" to Storage more options (FRM-439)
- Add stripe onboarding form link to prod settings page
- Disabled Thank You page for payments
- Add info box for payment amounts >$1000 (FRM-893)

**New Issues created (to track missing copy changes)**:

- FRM-960: Login page banner for payments
- FRM-961: Landing page banner for payments
- FRM-962: Building drawer tooltip for payments
- FRM-963: Custom Thank you page for payments

### Announcement rollout

Screenshot 2023-05-31 at 7 36 23 AM

#### Content

Screenshot 2023-05-31 at 7 36 30 AM

#### Badge

Screenshot 2023-05-31 at 7 40 44 AM

### Form creation - Storage mode option

Screenshot 2023-05-31 at 7 39 33 AM

### Prod env - Payment settings page infobox before connection

Screenshot 2023-05-31 at 7 40 22 AM

#### Storage-mode form with payments

Screenshot 2023-05-31 at 7 45 55 AM

#### Storage-mode form without payments and email-mode forms

Screenshot 2023-05-31 at 7 47 38 AM

### Infobox in Payment Input Panel

Screenshot 2023-05-31 at 7 48 50 AM

## Tests

- [ ] On a new login (or `has-seen-accouncement-rollout-...` cleared on localstorage), ensure that the announcement rollout modal shows with the payment feature.
- [ ] On a new login (or `users.flags.lastSeenFeatureUpdateVersion` < 2), ensure that:
   - [ ] There is a red badge beside "What's new".
   - [ ] "What's new" drawer shows with the payment feature.
- [ ] When creating a form, the Storage mode option should have "Supports payments" as one of the pointers.
- [ ] In the production env's payment settings page, check that the Stripe onboarding form link is added to the infobox.
- [ ] For the Thank You page,
   - [ ] Storage-mode form with payments
      - [ ] All customising fields are disabled .
      - [ ] Preview reflects the wording for thank you page for payments.
   - [ ] Storage-mode form without payments - has normal Thank You page as before.
   - [ ] Email-mode form - has normal Thank You page as before.
- [ ] Check that there is an infobox in the payment input panel for payment amounts of > $1000

### CODE REVIEW

File: frontend/src/features/admin-form/create/end-page/EndPageDrawer.tsx [hunk 1]
```diff
@@ -210,6 +225,10 @@ export const EndPageDrawer = (): JSX.Element | null => {
     ),
   )
 
+  const isPaymentEnabled =
+    form?.responseMode === FormResponseMode.Encrypt &&
+    form?.payments_field.enabled
```
Comment (LinHuiqing) [inline frontend/src/features/admin-form/create/end-page/EndPageDrawer.tsx:230]:
```suggestion
    form?.payments_channel.channel !== PaymentChannel.Unconnected
```
Will need to update this (along with import PaymentChannel) once the payment enablement PR is merged in

File: frontend/src/features/rollout-announcement/Announcements.tsx [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
+    description:
+      'Citizens can now pay for fees and services directly on your form! We integrate with Stripe to provide reliable payments and hassle-free reconciliations. Payment methods we support include credit card and PayNow.',
+    animationData: FirstAnnouncement,
+  },
   {
     title: 'Drag-and-drop form creation',
     description:
       'Add and rearrange fields with ease, and see a preview of your form while editing it.',
-    animationData: FirstAnnouncement,
+    animationData: SecondAnnouncement,
```
Comment (timotheeg) [inline frontend/src/features/rollout-announcement/Announcements.tsx]: /noChangeNeeded: we need a better way to do this than naming the assets with positional indicators 😑 . The order is defined in this file (in the array); the assets themselves should only be named on what they contain or are (e.g. `import dndAnnouncement from /assets/new-feature-dnd.json`)

File: frontend/src/features/whats-new/FeatureUpdateList.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import dedent from 'dedent'
 import { JsonObject, RequireExactlyOne } from 'type-fest'
 
-import Animation1 from './assets/1-dnd.json'
+import Animation1 from './assets/1-payments.json'
 import Animation2 from './assets/2-search-and-filter.json'
+import Animation3 from './assets/3-dnd.json'
```
Comment (timotheeg) [inline frontend/src/features/whats-new/FeatureUpdateList.ts:6]: Same comment on not naming assets with positions in their name, but also same comment that there's no need to change that now in the PR.

**Review by timotheeg — APPROVED** [review]
LGTM

**Comment by LinHuiqing** [comment]
Oh, by the way, I'll be setting the default max payment value as `Number.MAX_VALUE` because we are not imposing any max (and we have the infobox for amounts more than $1000).

File: frontend/src/features/rollout-announcement/Announcements.tsx (hunk 2 above)
Comment (LinHuiqing) [reply frontend/src/features/rollout-announcement/Announcements.tsx]: Done in [fb6edda](https://github.com/opengovsg/FormSG/pull/6407/commits/fb6eddae42738879705bcf910b7aa9d86d1ec369) along with removing one announcements!

File: src/app/config/features/payment.config.ts [hunk 4]
```diff
@@ -45,7 +45,7 @@ const paymentFeature: Schema<IStripe> = {
   maxPaymentAmountCents: {
     doc: 'Maximum amount that can be paid for a form',
     format: Number,
-    default: 100000, // $1000, due to IRAS limit for invoice
+    default: Number.MAX_VALUE,
```
Comment (timotheeg) [inline src/app/config/features/payment.config.ts]: LOL, let's have a default a bit saner than that 😅 .

```javascript
Number.MAX_VALUE
1.7976931348623157e+308
```

I'll just suggest arbitrarily 1M; but realistically, I'd rather see 100k or even 10k, and getting notified if anyone is collecting more than 1k 😰

Comment (LinHuiqing) [reply src/app/config/features/payment.config.ts]: Set it to 1M in [e5af962](https://github.com/opengovsg/FormSG/pull/6407/commits/e5af96257a3e27aad04365b00ce9bc774a11aba6), we can adjust this up/down by changing the env var anyway.

**Review by timotheeg — APPROVED** [review]
LGTM

**Comment by LinHuiqing** [comment]
Oh wait it just merged tho the Chromatic failed... Oh well.

**Pull request merged**